lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -116,12 +120,9 @@ def _on_relation_changed(self, event) -> None:
                 )
             logger.info("Skipping relation joined event: hook checks did not pass")
             return
-
         config_server_db = self.generate_config_server_db()
-
         # create user and set secrets for mongos relation
         self.charm.client_relations.oversee_users(None, None)
```

Comment:
The thing is that `oversee_users` is a reconcile pattern that handles user creation, user deletion, and many other things.
I'm not sure we can easily remove the `set_credentials` from this function (it's already embedded in another function doing many things).
I need to talk with @MiaAltieri about that to decide what we could do because it might imply bigger changes, and I'm not sure if we have situations were users are created in the `relation_joined` event (it would make it way easier if that never happens.
Also I know the oversee users function is also used elsewhere for another reason I forgot.

Reply:
Great suggestion @juditnovak ! Certainly appreciate your keen eyes on this PR 🙇 

Lets definitely do this, I want to be sure that we use the lib correctly. I think it is out of scope for this PR but I believe there is a ticket that @Gu1nness created that should address it. Maybe it can even been addressed in pairing session in the future pulse.

Reply:
No worries, I'm happy to help :-)

However, I'm struggling to approve the current solution. (I find it a particularly important matter that security-related workflows are followed strictly... Compared to which the current solution feels like a hack... :-( )

Is it OK if I just participated this PR as an "advisor", and approve on the one that's implementing the right thing?
(It should not block you from merging, you Guys can provide enough approvals w/o me.)

Reply:
It is completely OK to me.
As we talked about, I know this first solution is not sufficient to have a good and clear path, but at least it works.
I have already created a new ticket in order to have a better solution, and I hope to bring it in next pulse if I can, and I would love your advice on the next PR as well since you know the library and its ways, way better than I do :)